src/alternatives.js

```diff
@@ -0,0 +1,136 @@
+import inherits from './util/inherits';
+import MixedSchema from './mixed';
+import { alternatives as locale } from './locale.js';
+import isSchema from './util/isSchema';
+import typeOf from 'type-name';
+import { SynchronousPromise } from 'synchronous-promise';
+import createValidation from './util/createValidation';
+import printValue from './util/printValue';
+import runValidations from './util/runValidations';
+
+let promise = sync => sync ? SynchronousPromise: Promise;
+const invalidMarker = {};
+
+
+export default function AlternativesSchema() {
+    if (!(this instanceof AlternativesSchema))
+        return new AlternativesSchema()
+
+    MixedSchema.call(this, { type: 'alternatives' })
+}
+
+inherits(AlternativesSchema, MixedSchema, {
+
+    _validate(_value, options = {}) {
```

Comment:
most of this logic is copied from mixed, which you don't need to do. Check out how Array cast/validate is implemented because it's similar to what you want to do here, e.g. proxy the calls to a "sub schema"